src/app/src/test/kotlin/org/vaccineimpact/api/tests/controllers/ModellingGroupControllersTests.kt

```diff
@@ -214,4 +304,51 @@ class ModellingGroupControllersTests : ControllerTests<ModellingGroupController>
                     CoverageSet(1, "tId", "name", "vaccine", GAVISupportLevel.WITH, ActivityType.CAMPAIGN)
             )
     )
+
+    private fun generateCoverageRows(): List<LongCoverageRow>
+    {
+        val countries = listOf("ABC", "DEF")
+        val setNames = listOf("set1", "set2")
+        val vaccines = listOf("vaccine1", "vaccine2")
+        val supportLevels = listOf(GAVISupportLevel.WITHOUT, GAVISupportLevel.WITH)
+        val ageFroms = listOf(BigDecimal.ZERO, BigDecimal(5))
+        val ageTos = listOf(BigDecimal.TEN, BigDecimal(5))
+        val activityTypes = listOf(ActivityType.CAMPAIGN, ActivityType.NONE)
+
+        val listToReturn = mutableListOf<LongCoverageRow>()
+
+        for (country in countries)
+        {
+            for (set in setNames)
+            {
+                for (vaccine in vaccines)
+                {
+                    for (support in supportLevels)
+                    {
+                        for (ageFrom in ageFroms)
+                        {
+                            for (ageTo in ageTos)
+                            {
+                                for (activity in activityTypes)
+                                {
+                                    for (year in years)
+                                    {
+                                        listToReturn.add(LongCoverageRow("sId", set, vaccine, support, activity,
+                                                country, "$country-Name", year, ageFrom, ageTo, "$ageFrom-$ageTo",
+                                                random.nextDecimal(1000, 10000, numberOfDecimalPlaces = 2),
+                                                random.nextDecimal(1000, 10000, numberOfDecimalPlaces = 2)))
```

Comment:
I prefer to avoid truly random numbers in tests, as they can lead to non-deterministic test failures that are hard to track down. Maybe have an integer outside all the for loops that goes up by one for every row, and then use the current value of that to make your coverage/target values?

Reply:
You're right, this was lazy on my part. Have fixed.